### ISSUE TITLE
[Change proposal] [no-unused-vars] Passing a variable to a recursively called function shouldn't count as usage

### ISSUE THREAD

**Issue by nicolo-ribaudo** [eslint/eslint#9282]
* **ESLint Version:** 4.6.0

**Source code**

```js
function factorial(n, unused1, unused2) {
  return n < 1 ? 1 : n * factorial(n - 1, unused1);
}

factorial(4);
```

https://eslint.org/demo/#eyJ0ZXh0IjoiZnVuY3Rpb24gZmFjdG9yaWFsKG4sIHVudXNlZDEsIHVudXNlZDIsIHVudXNlZDMpIHtcbiAgdmFyIG0gPSB1bnVzZWQyO1xuICByZXR1cm4gbiA8IDEgPyAxIDogbiAqIGZhY3RvcmlhbChuIC0gMSwgdW51c2VkMSwgbSk7XG59XG5cbmZhY3RvcmlhbCg0KTsiLCJvcHRpb25zIjp7InBhcnNlck9wdGlvbnMiOnsiZWNtYVZlcnNpb24iOjcsInNvdXJjZVR5cGUiOiJtb2R1bGUiLCJlY21hRmVhdHVyZXMiOnt9fSwicnVsZXMiOnsibm8tdW51c2VkLXZhcnMiOjJ9LCJlbnYiOnt9fX0=

**What did you expect to happen?**
`unused1` is actually unused, since it is just passed to the function itself without being used in any other way, and thus it can be safely removed. I think `no-unused-vars` should warn about that.

**What actually happened? Please include the actual, raw output from ESLint.**
Only `unused2` was (correctly) reported as unused.

**Comment by not-an-aardvark** [comment]
Thanks for your interest in improving ESLint. Unfortunately, it looks like this issue didn't get consensus from the team, so I'm closing it. We define [consensus](http://eslint.org/docs/maintainer-guide/issues.html#consensus) as having three :+1:s from team members, as well as a team member willing to champion the proposal. This is a high bar by design -- we can't realistically accept and maintain every feature request in the long term, so we only accept feature requests which are useful enough that there is consensus among the team that they're worth adding.

Since ESLint is pluggable and can load custom rules at runtime, the lack of consensus among the ESLint team doesn't need to be a blocker for you using this in your project, if you'd find it useful. It just means that you would need to [implement the rule](http://eslint.org/docs/developer-guide/working-with-rules) yourself, rather than using a bundled rule that is packaged with ESLint.
